context/context.go

```diff
@@ -151,7 +152,7 @@ func NewEnv() (Env, error) {
 
 // NewContextWD creates a new context. It looks for a root folder by finding
 // a vendor file.
-func NewContextWD(rt RootType) (*Context, error) {
+func NewContextWD(base *Context, rt RootType) (*Context, error) {
```

Comment:
I would prefer to avoid this base Context thing.

Could we just "|" Or the RootType with a new RootType, like  `NewContextWD(RootVendor | RootExternal)`?